addons/project_timesheet_holidays/models/account_analytic.py

```diff
@@ -10,8 +10,27 @@ class AccountAnalyticLine(models.Model):
 
     holiday_id = fields.Many2one("hr.leave", string='Leave Request')
     global_leave_id = fields.Many2one("resource.calendar.leaves", string="Global Time Off", ondelete='cascade')
+    task_id = fields.Many2one(domain="[('company_id', '=', company_id), ('project_id.allow_timesheets', '=', True),"
+        "('project_id', '=?', project_id), ('is_timeoff_task', '=', False)]")
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_leave(self):
         if any(line.holiday_id for line in self):
-            raise UserError(_('You cannot delete timesheets linked to time off. Please, cancel the time off instead.'))
+            raise UserError(_('You cannot delete timesheets that are linked to time off requests. Please cancel your time off request from the Time Off application instead.'))
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if not self.env.su:
+            task_list = []
+            for vals in vals_list:
+                if vals.get('task_id'):
+                    task_list.append(vals['task_id'])
+            tasks = self.env['project.task'].search([('id', 'in', task_list)], limit=2)
+            if any(tasks.mapped('is_timeoff_task')):
```

Comment:
Here is what I though of 🙂, adapted to the search_count as now it also accepts a limit argument 🤩. 
```suggestion
            task_ids = [vals['task_id'] for vals in vals_list if vals.get('task_id')]
            has_timeoff_task = self.env['project.task'].search_count([('id', 'in', task_ids), ('is_timeoff_task', '=', True)], limit=1) > 0
            if has_timeoff_task:
```

Reply:
Done! Thanks :slightly_smiling_face: